### ISSUE TITLE
As a farmer, be able to handle payments from users in different VAT regimes

### ISSUE THREAD

**Issue by gmachtel** [threefoldtech/jumpscaleX_threebot#126]
As a farmer you get requests from users which operate in different VAT regions. 

[Business requirement](https://docs.google.com/document/d/1Cz-eMEe3oL85LepD542ttDp3avwOMnQ0t52fGu0Ynr0/edit#)

**Assumptions**

**More info**

A user will be invoiced for his capacity reservation, but will have to pay a VAT amount which is depending on the country he is residing (different rules apply for EU and non-EU). 
VAT API to be called in order to be able to determine the correct VAT for a specific country. 
API that appears to contain this info : [VATsense](https://vatsense.com)

**DOD**
- Country of the user registered in the user's Threebot
- Farmer's invoice takes this country into account at invoice creation time
- Farmer is able to configure his wallets so that VAT are split up per country (if he wishes)

**TODO**
[] Configuration screen for farmer to define his required TFT wallets
[] Call API to get the right VAT for the right customer

**Comment by robvanmieghem** [comment]
@gmachtel Ok, VATt when using Fiat is simple, how do we charge Vat when paying with TFT,s ?
I propose we just implement the VAT rules in TFT and leave it up to the farmer how he should declare the income in his legislation.

**Comment by gmachtel** [comment]
I agree to leave it up to the farmer. VAT when using TFT is no different than fiat (legal steps are ongoing to consider TFT as a payment currency). 
VAT is to be expressed in TFT as well, maybe additional step is indication of exchange rate with EUR or USD at moment of the transaction. 
From technical perspective, VAT is 'simple' but the process is not simple for a farmer as it can become quite a mess for the farmer (both in fiat and in TFT). The APIs I mentioned are probably fit for purpose to get the % (didn't check yet which one fits best), but we might need to help the farmer with different wallet address. That's what Andreas suggested.

**Comment by robvanmieghem** [comment]
Another option might be to just ignore it from our side and let the farmer announce it including VAT even though there might be a difference to where the user resides even though the farmer would have to create invoices then.

**Comment by gmachtel** [comment]
Not sure we don't have legal obligations providing the right info on VAT to the grid user. As payment is done, there is an invoice system needed. If we don't register this on the system, the farmer has no way to send out invoices in a decent manner. I'll talk to Andreas about it. 
Next to that there is also income for TF Foundation (10%) and TF Tech (10% if certified capacity), also subject to VAT.

**Comment by andhartl** [comment]
think we need to give the farmer and user the option to get vat correct invoices for their book keeping. on noway of looking at this is that if we do not over the invoicing incl vat handling a lot of businesses - farmers and capacity users - will simple not be able to use capacity from the grid. hope this helps
